src/components/MobileActionBar.vue

```diff
@@ -38,14 +42,24 @@ export default defineComponent({
         }
 
         function receive() {
-            context.root.$router.push(nimOrBtc<string>('/receive', '/btc-receive'));
+            if (width.value <= 700 /* Full mobile breakpoint */ && activeMobileColumn.value !== ColumnType.ADDRESS) {
+                // redirect to the address selector
+                context.root.$router.push('/receive');
+            } else {
+                context.root.$router.push(nimOrBtc<string>('/receive/nim', '/receive/btc'));
```

Comment:
The `<string>` specification of the generic type of the `nimOrBtc` function is actually not necessary, Typescript figures it out itself.